packages/odyssey-react-mui/src/labs/DataFilters.tsx

```diff
@@ -175,22 +175,46 @@ const DataFilters = ({
     }
   }, [onChangeSearch, searchValue, searchDelayTime, hasSearchSubmitButton]);
 
-  const handleInputChange = useCallback(
-    (filterId: string, value: DataFilterValue, submit: boolean = false) => {
+  const updateInputValue = useCallback(
+    ({ filterId, value }: { filterId: string; value: DataFilterValue }) => {
       setInputValues({ ...inputValues, [filterId]: value });
+    },
+    [inputValues]
+  );
 
-      if (submit) {
-        const updatedFilters = filtersProp.map((filter) => ({
-          ...filter,
-          value: filter.id === filterId ? value : inputValues[filter.id],
-        }));
+  const updateFilters = useCallback(
+    ({ filterId, value }: { filterId: string; value: DataFilterValue }) => {
+      const updatedFilters = filtersProp.map((filter) => ({
+        ...filter,
+        value: filter.id === filterId ? value : inputValues[filter.id],
+      }));
 
-        setFilters(updatedFilters);
-      }
+      setFilters(updatedFilters);
     },
     [inputValues, filtersProp]
   );
 
+  const getAutoCompleteLabel = <
+    Value extends { label: string } | Array<{ label: string }>
+  >(
+    value: Value
+  ) => {
+    if (Array.isArray(value)) {
+      // Iterating to find the label
+      return value
+        .map((valueElement) => {
+          if (typeof valueElement !== "string") {
+            return valueElement.label;
+          }
+
+          return;
```

Comment:
I'm surprised this doesn't throw a typescript error because we're not returning the same thing. Either we're returning a `string` or `undefined`.

I wonder if it would be better to say "if it's a string, then `return undefined`; otherwise, return `valueElement.label`". I say that because I prefer the positive over the negative with `if` statements:
```suggestion
          if (typeof valueElement === "string") {
            return undefined;
          }

          return valueElement.label;
```